src/setup/node.rs

```diff
@@ -36,11 +37,11 @@ impl Node {
         })
     }
 
-    pub async fn start_with_peers(peers: Vec<SocketAddr>) -> Result<Self> {
-        let mut node = Self::new()?;
+    pub async fn start(path: PathBuf, ip_addr: IpAddr, peers: Vec<SocketAddr>) -> Result<Self> {
```

Comment:
we don't need `new` and `start`, as they do the same thing now

Reply:
It seems like ideally there should be some `NodeBuilder`. As it is now, we have some functions that must be called before start, but type system doesn't prevent it :-/

Reply:
Refactored to builder pattern.